src/renderer/rendered_window.rs

```diff
@@ -684,6 +670,63 @@ impl RenderedWindow {
         self.scroll_delta = 0;
     }
 
+    fn map_lines<F>(&self, mut f: F)
```

Comment:
Theres a todo here that needs replaced. But also I think this function should be rewritten and renamed to iter_lines. It should look something like this:
```rust
fn iter_lines<'a>(&'a self) -> Box<dyn Iterator<Item = &'a Rc<RefCell<Line>>> + 'a> {
    let scroll_offset_lines = self.scroll_animation.position.floor() as isize;
    let height = self.grid_size.height as isize;
    if height == 0 {
        return Box::new(std::iter::empty());
    }

    let top_border_lines = self
        .actual_lines
        .iter_range(0..self.viewport_margins.top as isize)
        .flatten();
    let actual_line_count = self.actual_lines.len() as isize;
    let bottom_border_lines = self
        .actual_lines
        .iter_range(
            actual_line_count - self.viewport_margins.bottom as isize..actual_line_count,
        )
        .flatten();

    if self.scrollback_lines.is_empty() {
        Box::new(top_border_lines.chain(bottom_border_lines))
    } else {
        let scrollable_lines = self
            .scrollback_lines
            .iter_range(scroll_offset_lines..scroll_offset_lines + height + 1)
            .flatten();

        Box::new(
            top_border_lines
                .chain(scrollable_lines)
                .chain(bottom_border_lines),
        )
    }
}
```

Bonus points if another similar function is introduced called iter_scrollback_lines which is identical but targets the scrollback_lines rather than actual lines and can be used in the prepare_lines function

Reply:
Update!